#! /usr/bin/env python

PACKAGE='kcl_sbc'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
gen.add("frequency", double_t,    0,    "Frequency.", 1, 0.1,	10.1)
gen.add("mean",       double_t,    0,    "Mean", 1,     0, 10)
gen.add("std_dev",       double_t,    0,    "StdDev", 0.1,     0, 10.1)

#size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                       gen.const("Medium",     int_t, 1, "A medium constant"),
#                       gen.const("Large",      int_t, 2, "A large constant"),
#                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
#                     "An enum to set size")
#gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "kcl_sbc", "P_Reg"))
